src/scheduler/MpiMessageBuffer.cpp

```diff
@@ -0,0 +1,69 @@
+#include <faabric/scheduler/MpiMessageBuffer.h>
+#include <faabric/util/logging.h>
+
+namespace faabric::scheduler {
+typedef std::list<MpiMessageBuffer::Arguments>::iterator ArgListIterator;
```

Comment:
As mentioned re. the naming of `Arguments`, I think the name of this `ArgListIterator` is a little ambiguous. Given that it's an `MpiMessageBuffer` it would perhaps be clearer if it was something like `MpiMessageIterator`